### PR TITLE
Document OpenShift Install Logs in S3 Enhancement.

### PR DESCRIPTION
**Pull request by twiest**

### CODE REVIEW

File: docs/enhancements/install_logs.md
Comment (dgoodwin) [inline docs/enhancements/install_logs.md:79]: An additional drawback, presently we assume a default of gathering logs and putting on pvcs. If you don't have quota/storage classes installs will fail to launch until you explicitly go setup HiveConfig to skipGatherLogs.

In this new mode it sounds more like our default is to not gather and this is a bit of a sketchy thing to roll out on people. I don't want to maintain the pv code as an alternative method though, so I think it's unavoidable. Perhaps worth noting the drawback here though that we're shifting from a default of gather to a default of don't gather.

FWIW ACM disabled log gathering in all their installs due to the pvc issues.

**Review by dgoodwin — CHANGES_REQUESTED** [review]
Added my thoughts. We'll want to put this infront of ACM and SRE-P once we're happy but before we merge to make sure they're on board.

**Comment by twiest** [comment]
> Added my thoughts. We'll want to put this infront of ACM and SRE-P once we're happy but before we merge to make sure they're on board.

Added a hold so that it won't merge until we've put it in front of ACM and SRE-P.

File: docs/enhancements/install_logs.md
Comment (akhil-rane) [reply docs/enhancements/install_logs.md:63]: expiration times can be set on minio buckets now check [this](https://docs.min.io/docs/minio-bucket-lifecycle-guide.html)

File: docs/enhancements/install_logs.md
Comment (dgoodwin) [reply docs/enhancements/install_logs.md:28]: I wish we could but no we have a bit of a rule here that is hard to explain, we have two layers of operator here, hive itself (controllers, hiveadmission) is an operator, and then we have the hive-operator which deploys hive. HiveConfig is for the hive-operator to know how to deploy hive, so we never read it in hive-controllers itself, we always pass any config through via some other means. It's unfortunate and awkward and I'm not sure it's a great idea, but it's what we do today.

File: docs/enhancements/install_logs.md
Comment (staebler) [reply docs/enhancements/install_logs.md:62]: I agree. One scenario where it would be helpful to keep the install logs is for failures of the CI tests that run on stage and prod. Currently, if those installs fail, we have a short window of time to capture the logs before the test ends. It would be nice if those logs persisted for longer.

File: docs/enhancements/install_logs.md
Comment (staebler) [reply docs/enhancements/install_logs.md:79]: Are we planning to deprecate `skipGatherLogs`?

File: docs/enhancements/install_logs.md
Comment (twiest) [reply docs/enhancements/install_logs.md:28]: Added.

File: docs/enhancements/install_logs.md
Comment (twiest) [reply docs/enhancements/install_logs.md:67]: Added.

**Comment by twiest** [comment]
@dgoodwin @akhil-rane @staebler Thanks for the feedback. I believe I've addressed them. PTAL.

File: docs/enhancements/install_logs.md
Comment (gregsheremeta) [inline docs/enhancements/install_logs.md:34]: s/blog/blob/

File: docs/enhancements/install_logs.md
Comment (gregsheremeta) [reply docs/enhancements/install_logs.md]: /cc @jeefy 
we talked about persisting these logs in CICD.
what do you need to be able to access these?

**Comment by gregsheremeta** [comment]
/cc @igoihman @nimrodshn @zvikorn @jnpacker @TheRealHaoLiu

please review

File: docs/enhancements/install_logs.md
Comment (jeefy) [reply docs/enhancements/install_logs.md]: We have an AWS account that we use to push metrics to an S3 bucket. I think all we'd really need is the S3 bucket/file path and read access to that bucket.

Comment (twiest) [reply docs/enhancements/install_logs.md]: That sounds good.

How this works is that Hive is given creds and the url to the S3 bucket where to store them. We then put the URL of what we uploaded to our clusterprovision object.

So if someone has readonly creds to the bucket we were given and has access to the clusterprovision object, then they should be able to get the logs.

**Comment by twiest** [comment]
@mwoodson @jewzaam @jharrington22  This is the design doc for the proposed changes to how we handle install logs. PTAL.

**Comment by mwoodson** [comment]
I know that cleaning up the logs is left as configuration excercise of the S3 bucket by the admin, but i think some guidelines or suggestions would be useful.  This is something SREP, CICD, SDA can all help with.

Right now, if I were to set up this bucket, I would have a hard time knowing some "sane" defaults to set this up.  Even if we start with the current PV policy, this would be useful.

**Comment by twiest** [comment]
@mwoodson I believe I've addressed your feedback. PTAL.

**Comment by mwoodson** [comment]
@twiest thanks for the update, this helps out a lot!

**Comment by twiest** [comment]
@dgoodwin @staebler can I get an lgtm please?

File: docs/enhancements/install_logs.md
Comment (jewzaam) [inline docs/enhancements/install_logs.md:53]: Does the clusterprovisionname include the install attempt #? Do we store install logs for failed installs?

Comment (dgoodwin) [reply docs/enhancements/install_logs.md:53]: They do include the attempt number, ex: dgood-dev-0-vm2hj. We definitely will store the logs from failed attempts.

**Comment by twiest** [comment]
@dgoodwin thanks for the +1. I needed to rebase it for the merge.

Would you mind adding the +1 again please?

Also, would you mind closing your "change request"?  It's still showing you as having 1 "requested changes".

Thanks!

**Review by dgoodwin — APPROVED** [review]

**Pull request merged**
